pandas/io/tests/test_parsers.py

```diff
@@ -2690,6 +2690,41 @@ def test_uneven_lines_with_usecols(self):
         df = self.read_csv(StringIO(csv), usecols=usecols)
         tm.assert_frame_equal(df, expected)
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2]  # ['X']
+
+        def check_usecols(stringIO, expected):
+            df = self.read_csv(stringIO, names=names, usecols=usecols)
+            tm.assert_frame_equal(df, expected)
+
+        # first, check to see that the response of
+        # parser when faced with no provided columns
+        # throws the correct error, with or without usecols
+        errmsg = "No columns to parse from file"
+
+        with tm.assertRaisesRegexp(ValueError, errmsg):
```

Comment:
Similar to your comment above, the C engine raises a generic `ValueError`, while the Python parser would now raise an `EmptyDataError`.